implementations/micrometer-registry-dynatrace/src/test/java/io/micrometer/dynatrace/v2/DynatraceExporterV2Test.java

```diff
@@ -866,6 +867,40 @@ void shouldAddMetadataOnlyWhenUnitOrDescriptionIsPresent() {
                 "#gauge.du gauge dt.meta.description=temperature,dt.meta.unit=kelvin")));
     }
 
+    @Test
+    void shouldHaveUcumCompliantUnits() {
+        HttpSender.Request.Builder builder = spy(HttpSender.Request.build(config.uri(), httpClient));
+        when(httpClient.post(anyString())).thenReturn(builder);
+
+        meterRegistry.timer("test.timer").record(Duration.ofMillis(12));
+        meterRegistry.more().timeGauge("test.tg", Tags.empty(), this, TimeUnit.MICROSECONDS, x -> 1_000);
+        FunctionTimer.builder("test.ft", this, x -> 1, x -> 100, MILLISECONDS).register(meterRegistry);
+        Counter.builder("test.second").baseUnit("second").register(meterRegistry).increment(100);
+        Counter.builder("test.seconds").baseUnit("seconds").register(meterRegistry).increment(10);
+        FunctionCounter.builder("process.cpu.time", this, x -> 1_000_000).baseUnit("ns").register(meterRegistry);
+
+        Sample sample = meterRegistry.more().longTaskTimer("test.ltt").start();
+        clock.add(config.step().plus(Duration.ofSeconds(2)));
+
+        exporter.export(meterRegistry.getMeters());
+        sample.stop();
+
+        verify(builder).withPlainText(assertArg(body -> assertThat(body.split("\n")).containsExactlyInAnyOrder(
+                "test.timer,dt.metrics.source=micrometer gauge,min=12,max=12,sum=12,count=1 " + clock.wallTime(),
+                "#test.timer gauge dt.meta.unit=ms", "test.tg,dt.metrics.source=micrometer gauge,1 " + clock.wallTime(),
+                "#test.tg gauge dt.meta.unit=ms",
+                "test.ft,dt.metrics.source=micrometer gauge,min=100,max=100,sum=100,count=1 " + clock.wallTime(),
+                "#test.ft gauge dt.meta.unit=ms",
+                "test.second,dt.metrics.source=micrometer count,delta=100 " + clock.wallTime(),
+                "#test.second count dt.meta.unit=s",
+                "test.seconds,dt.metrics.source=micrometer count,delta=10 " + clock.wallTime(),
+                "#test.seconds count dt.meta.unit=s",
+                "process.cpu.time,dt.metrics.source=micrometer count,delta=1000000 " + clock.wallTime(),
+                "#process.cpu.time count dt.meta.unit=ns",
+                "test.ltt,dt.metrics.source=micrometer gauge,min=62000,max=62000,sum=62000,count=1 " + clock.wallTime(),
+                "#test.ltt gauge dt.meta.unit=ms")));
+    }
+
```

Comment:
I wrote another test that ensures that all the TimeGauge instruments are mapped to the same `ms` base unit, but I only now realized that it's basically testing the same thing five times. Feel free to ignore this; I posted it anyway because it showcases how the TimeGauge would work with the Dynatrace exporter. 

```suggestion

    @Test
    void timeGaugeUnitIsMappedToBaseUnit() {
        HttpSender.Request.Builder builder = spy(HttpSender.Request.build(config.uri(), httpClient));
        when(httpClient.post(anyString())).thenReturn(builder);

        TimeGauge.builder("tg1", () -> 1., MILLISECONDS).register(meterRegistry);
        TimeGauge.builder("tg2", () -> 2., MICROSECONDS).register(meterRegistry);
        TimeGauge.builder("tg3", () -> 3., NANOSECONDS).register(meterRegistry);
        TimeGauge.builder("tg4", () -> 4., SECONDS).register(meterRegistry);
        TimeGauge.builder("tg5", () -> 5., MINUTES).register(meterRegistry);

        clock.add(config.step());
        exporter.export(meterRegistry.getMeters());

        // since the base unit of the registry is millis, everything will be exported as millis
        verify(builder).withPlainText(assertArg(body -> {
            assertThat(body.split("\n")).containsExactlyInAnyOrder(
                "tg1,dt.metrics.source=micrometer gauge,1 " + clock.wallTime(),
                "#tg1 gauge dt.meta.unit=ms",
                "tg2,dt.metrics.source=micrometer gauge,0.002 " + clock.wallTime(),
                "#tg2 gauge dt.meta.unit=ms",
                "tg3,dt.metrics.source=micrometer gauge,3.0E-6 " + clock.wallTime(),
                "#tg3 gauge dt.meta.unit=ms",
                "tg4,dt.metrics.source=micrometer gauge,4000 " + clock.wallTime(),
                "#tg4 gauge dt.meta.unit=ms",
                "tg5,dt.metrics.source=micrometer gauge,300000 " + clock.wallTime(),
                "#tg5 gauge dt.meta.unit=ms"
                );
        }));
    }
```